geomesa-convert/geomesa-convert-common/src/main/scala/org/locationtech/geomesa/convert/Transformers.scala

```diff
@@ -482,5 +481,43 @@ class MapListFunctionFactory extends TransformerFunctionFactory {
         }.toMap.asJava
     }
   }
-
+}
+class CastFunctionFactory extends TransformerFunctionFactory {
+  override def functions = Seq(string2double,string2int,string2float,string2long,string2boolean)
+
+  val string2double = TransformerFn("string2double") {
+    args => {
+      if (args(0) == null || args(0).asInstanceOf[String] == "") {
+        if (args(1) == null || args(1).asInstanceOf[String] == "") args(1) else args(1).asInstanceOf[String].toDouble
+      } else args(0).asInstanceOf[String].toDouble
+    }
+  }
+  val string2int = TransformerFn("string2int") {
+    args => {
+      if (args(0) == null || args(0).asInstanceOf[String] == "") {
+        if (args(1) == null || args(1).asInstanceOf[String] == "") args(1) else args(1).asInstanceOf[String].toInt
+      } else args(0).asInstanceOf[String].toInt
+    }
+  }
+  val string2float = TransformerFn("string2float") {
```

Comment:
+1 for overall review. I'm not really in favor of names like '2whatever' - I would advocate for fully spelled out camel-case names for consistency. E.g. ```stringLength``` (or just ```length```) instead of ```strlen```, etc. We could always keep abbreviated synonyms around too.